lib/SILGen/SILGenPattern.cpp

```diff
@@ -738,15 +738,17 @@ forwardIntoSubtree(SILGenFunction &SGF, SILLocation loc,
   ManagedValue outerMV = outerCMV.getFinalManagedValue();
   if (!outerMV.hasCleanup()) return outerCMV;
 
-  assert(outerCMV.getFinalConsumption() != CastConsumptionKind::CopyOnSuccess
-         && "copy-on-success value with cleanup?");
+  assert(outerCMV.getFinalConsumption() != CastConsumptionKind::CopyOnSuccess &&
```

Comment:
Might be better to instead assert that it is one of TakeAlways or TakeOnSuccess?

Reply:
I am fine with this.